java-checks/src/main/java/org/sonar/java/checks/SimpleClassNameCheck.java

```diff
@@ -0,0 +1,127 @@
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks;
+
+import com.google.common.collect.ImmutableList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.sonar.check.Rule;
+import org.sonar.java.model.SyntacticEquivalence;
+import org.sonar.java.resolve.SemanticModel;
+import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
+import org.sonar.plugins.java.api.JavaFileScannerContext;
+import org.sonar.plugins.java.api.semantic.Symbol;
+import org.sonar.plugins.java.api.tree.IdentifierTree;
+import org.sonar.plugins.java.api.tree.ImportTree;
+import org.sonar.plugins.java.api.tree.MemberSelectExpressionTree;
+import org.sonar.plugins.java.api.tree.Tree;
+import org.sonar.plugins.java.api.tree.Tree.Kind;
+
+@Rule(key = "S1942")
+public class SimpleClassNameCheck extends IssuableSubscriptionVisitor {
+
+  private static final String MESSAGE = "Replace this fully qualified name with \"%s\"";
+  private Set<Tree> hasIssue = new HashSet<>();
+  private Set<Tree> importedWildcards = new HashSet<>();
+
+  @Override
+  public List<Tree.Kind> nodesToVisit() {
+    return ImmutableList.of(Kind.IMPORT, Kind.MEMBER_SELECT);
+  }
+
+  @Override
+  public void scanFile(JavaFileScannerContext context) {
+    super.scanFile(context);
+    hasIssue.clear();
+    importedWildcards.clear();
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+    if (!hasSemantic()) {
+      return;
+    }
+
+    if (tree.is(Kind.IMPORT)) {
+      visitImport((ImportTree)tree);
+
+    } else {
+      visitMemberSelect((MemberSelectExpressionTree)tree);
+    }
+  }
+
+  private void visitMemberSelect(MemberSelectExpressionTree tree) {
+    if (!isInImport(tree) && !hasIssue.contains(tree) && !tree.identifier().symbol().isPackageSymbol()) {
+
+      for (Tree importedWildcard : importedWildcards) {
+        if (SyntacticEquivalence.areEquivalent(importedWildcard, tree.expression())) {
+          reportIssue(tree, String.format(MESSAGE, tree.identifier().name()));
+          return;
+        }
+
+      }
+    }
+
+  }
+
+  private static boolean isInImport(Tree tree) {
+    Tree parent = tree.parent();
+
+    while (parent != null) {
+      if (parent.is(Kind.IMPORT)) {
+        return true;
+      }
+      parent = parent.parent();
+    }
+
+    return false;
+  }
+
+  private void visitImport(ImportTree tree) {
+    SemanticModel semanticModel = (SemanticModel) context.getSemanticModel();
+    Symbol symbol = semanticModel.getSymbol(tree);
+
+    if (symbol != null) {
+      checkImportedSymbol(symbol);
+
+    } else {
+      if (tree.qualifiedIdentifier().is(Kind.MEMBER_SELECT)) {
+        MemberSelectExpressionTree qualifiedIdentifier = (MemberSelectExpressionTree)tree.qualifiedIdentifier();
+        if (qualifiedIdentifier.identifier().name().equals("*")) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Move the "*" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1132)